File: app/src/main/java/com/jermaine/awesomelibraries/di/component/AppComponent.java
package com.jermaine.awesomelibraries.di.component;

import com.jermaine.awesomelibraries.di.module.AppModule;
import com.jermaine.awesomelibraries.repository.local.RepositoryLocalImpl;
import com.jermaine.awesomelibraries.repository.server.RepositoryServerImpl;
import com.jermaine.awesomelibraries.view.activity.additem.AddRepoPresenterImpl;
import com.jermaine.awesomelibraries.view.activity.repolist.RepoListPresenterImpl;
import com.jermaine.awesomelibraries.view.activity.viewitem.ViewItemPresenterImpl;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {
    void inject(RepoListPresenterImpl repoListPresenter);

    void inject(RepositoryLocalImpl retrofitRepositoryLocal);

    void inject(RepositoryServerImpl retrofitRepositoryServer);

    void inject(ViewItemPresenterImpl viewItemPresenter);

    void inject(AddRepoPresenterImpl addRepoPresenter);
}

File: app/src/main/java/com/jermaine/awesomelibraries/di/component/AppComponent.kt
package com.jermaine.awesomelibraries.di.component

import com.jermaine.awesomelibraries.di.module.AppModule
import com.jermaine.awesomelibraries.repository.ReposRepositoryImpl
import com.jermaine.awesomelibraries.uicontroller.activity.additem.AddRepoViewModel
import com.jermaine.awesomelibraries.uicontroller.activity.repolist.RepoListActivity
import com.jermaine.awesomelibraries.uicontroller.activity.repolist.RepoListViewModel
import com.jermaine.awesomelibraries.uicontroller.activity.viewitem.ViewItemViewModel
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun inject(retrofitRepositoryServer: ReposRepositoryImpl)

    fun inject(repoListActivity: RepoListActivity)

    fun inject(repoListViewModel: RepoListViewModel)

    fun inject(viewItemViewModel: ViewItemViewModel)

    fun inject(addRepoViewModel: AddRepoViewModel)
}